diesel/src/sqlite/expression/helper_types.rs

```diff
@@ -27,3 +28,8 @@ pub type json_pretty<E> = super::functions::json_pretty<SqlTypeOf<E>, E>;
 #[cfg(feature = "sqlite")]
 pub type json_pretty_with_indentation<J, I> =
     super::functions::json_pretty_with_indentation<SqlTypeOf<J>, J, I>;
+
+/// Return type of [`json_valid(json)`](super::functions::json_valid())
+#[allow(non_camel_case_types)]
+#[cfg(feature = "sqlite")]
+pub type json_valid<E> = super::functions::json_valid<SqlTypeOf<E>, Integer>;
```

Comment:
```suggestion
pub type json_valid<E> = super::functions::json_valid<SqlTypeOf<E>, E>;
```

That needs to be just `E` and not `Integer` as the helper type accepts one generic argument for the SQL type (the `SqlTypeOf<E>` part) and one generic argument for the expression itself (the `E`))